pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -271,7 +283,13 @@ func (r *repository) Get(ctx context.Context, dgst digest.Digest, options ...dis
 	}
 
 	ref := imageapi.DockerImageReference{Namespace: r.namespace, Name: r.name, Registry: r.registryAddr}
-	manifest, err := r.manifestFromImageWithCachedLayers(image, ref.DockerClientDefaults().Exact())
+	if managed := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; managed == "true" {
+		ref.Registry = ""
```

Comment:
functions like `identifyCandidateRepositories` compare `ref.Registry` against a known localRegistry:
```
// skip anything that matches the innate registry
// TODO: there may be a better way to make this determination
if len(localRegistry) != 0 && localRegistry == ref.Registry {
	continue
}
```

should we really be clearing this, or setting it to a "local registry" value? the effects of this change are very hard to reason about